firewood/src/merkle/node/branch.rs

```diff
@@ -61,21 +61,7 @@ impl BranchNode {
     pub const MAX_CHILDREN: usize = MAX_CHILDREN;
     pub const MSIZE: usize = Self::MAX_CHILDREN + 2;
 
-    pub fn new(
```

Comment:
This function would've been `const` after this change so I just removed it. I think it's a bit silly to have a `new` function that just puts its parameters into fields. Harder to read that way imo.